psamm/metabolicmodel.py

```diff
@@ -174,11 +196,11 @@ def database(self):
 
     @property
     def reactions(self):
-        return iter(self._reaction_set)
+        return self._reaction_set
 
     @property
     def compounds(self):
-        return iter(self._compound_set)
+        return self._compound_set
```

Comment:
`compounds` is changed to return the private `set` which is mutable. This allows the caller to accidentally modify the set of compounds which is bad. Instead an immutable view on the set should be returned.